cfme/infrastructure/provider/__init__.py

```diff
@@ -87,15 +101,280 @@
 mon_btn = partial(tb.select, 'Monitoring')
 
 
-class InfraProviderTimelinesView(TimelinesView, BaseLoggedInPage):
+class InfraProvidersView(BaseLoggedInPage):
+    @property
+    def is_displayed(self):
+        return self.logged_in_as_current_user and \
+            self.navigation.currently_selected == ['Compute', 'Infrastructure', 'Providers'] and \
+            self.entities.title.text == 'Infrastructure Providers'
+
+    @View.nested
+    class toolbar(ProviderToolBar):  # NOQA
+        pass
+
+    @View.nested
+    class sidebar(ProviderSideBar):  # NOQA
+        pass
+
+    @View.nested
+    class entities(ProviderEntities):  # NOQA
+        pass
+
+    @View.nested
+    class paginator(PaginationPane):  # NOQA
+        pass
+
+
+class InfraProviderDetailsSummaryView(View):
+    properties = SummaryTable(title="Properties")
+    status = SummaryTable(title="Status")
+    relationships = SummaryTable(title="Relationships")
+    overview = SummaryTable(title="Overview")
+    smart_management = SummaryTable(title="Smart Management")
+
+
+class InfraProviderDetailsDashboardView(View):
+    # todo: need to develop this page
+    pass
+
+
+class InfraProviderDetailsView(BaseLoggedInPage):
+    title = Text('//div[@id="main-content"]//h1')
+    breadcrumb = BreadCrumb(locator='//ol[@class="breadcrumb"]')
+
+    @View.nested
+    class toolbar(DetailsProviderToolBar):  # NOQA
+        pass
+
+    @View.nested
+    class contents(View):
+        pass
+
+    def __getattribute__(self, item):
+        if item == 'contents':
+            view_type = self.toolbar.view_selector.selected
+
+            if view_type == 'Summary View':
+                return InfraProviderDetailsSummaryView(parent=self)
+
+            elif view_type == 'Dashboard View':
+                return InfraProviderDetailsDashboardView(parent=self)
+
+            else:
+                raise Exception('The form for provider with such name '
+                                'is absent: {}'.format(self.prov_type.text))
+        else:
+            return super(InfraProviderDetailsView, self).__getattribute__(item)
 
     @property
     def is_displayed(self):
+        subtitle = 'Summary' if self.toolbar.view_selector.selected == 'Summary View' \
+            else 'Dashboard'
+        title = '{name} ({subtitle})'.format(name=self.context['object'].name, subtitle=subtitle)
+
+        return self.logged_in_as_current_user and \
+            self.navigation.currently_selected == ['Compute', 'Infrastructure', 'Providers'] and \
+            self.breadcrumb.active_location == title
+
+
+class InfraProvidersDiscoverView(BaseLoggedInPage):
+    vcenter = Checkbox('discover_type_virtualcenter')
+    mscvmm = Checkbox('discover_type_scvmm')
+    rhevm = Checkbox('discover_type_rhevm')
+
+    from_ip1 = Input('from_first')
```

Comment:
honestly, I don't like parsing ip address and filling several separate fields. 
I'm thinking of creating one IPv4/IPv6 control which would do this for us.